function_git.go

```diff
@@ -8,9 +8,11 @@ import (
 )
 
 type Git struct {
-	URL        *string `yaml:"url,omitempty"`
-	Revision   *string `yaml:"revision,omitempty"`
-	ContextDir *string `yaml:"contextDir,omitempty"`
+	URL             *string `yaml:"url,omitempty"`
+	Revision        *string `yaml:"revision,omitempty"`
+	ContextDir      *string `yaml:"contextDir,omitempty"`
+	SSHSecret       *string `yaml:"sshSecret,omitempty"`
```

Comment:
As mentioned on the issue, it might be actually better if we create secrets for users.

Reply:
It will definitely be better and the UX is where I need some help. 

The secret will need to have the private key required for the repo. But to scrape the user's `.ssh` directory and automatically copying keys feels wrong. 

So, instead of `sshSecret` is it sufficient to have `sshPrivateKey` where the user can specify the file path?

Reply:
@zroubalik @gab-satchi for what I understand this secret will contain the tokens for the git repo credentials right? 
If this is only supposed to work with on-cluster builds, right? If so, I would follow the same approach used by tekton where the user is responsible for creating a secret that will be used to connect to a container registry. if this is only relevant for on cluster build I consider this to be a platform configuration more than something that the developer should worry about.

Reply:
> for what I understand this secret will contain the tokens for the git repo credentials right?

That's right, with the current implementation, the user will need to either create an `sshSecret` with the private key contents or a `basicAuthSecret` for basic auth (not recommended for security reasons).

> if this is only relevant for on cluster build I consider this to be a platform configuration more than something that the developer should worry about.

One requirement is that the secret needs to be in the same namespace as the function. Would that still be considered a 'platform' configuration?